ui/episode/track/src/main/java/app/tivi/episode/track/EpisodeTrackPresenter.kt

```diff
@@ -48,17 +51,30 @@ import me.tatarka.inject.annotations.Assisted
 import me.tatarka.inject.annotations.Inject
 
 @Inject
-class EpisodeTrackViewModel(
-    @Assisted savedStateHandle: SavedStateHandle,
+class EpisodeTrackUiPresenterFactory(
+    private val presenterFactory: (EpisodeTrackScreen, Navigator) -> EpisodeTrackPresenter,
+) : Presenter.Factory {
+    override fun create(
+        screen: Screen,
+        navigator: Navigator,
+        context: CircuitContext,
+    ): Presenter<*>? = when (screen) {
```

Comment:
Side not - these are `fun interface` so could it be simpler to use the function syntax?

Reply:
ehh nevermind I guess it's not possible to use class delegation with a function input. Tried it locally without success